python/graphstorm/gconstruct/transform.py

```diff
@@ -255,12 +282,152 @@ def process_labels(data, label_confs, is_node):
         train_mask[train_idx] = 1
         val_mask[val_idx] = 1
         test_mask[test_idx] = 1
-    if label_col is None:
-        return {'train_mask': train_mask,
-                'val_mask': val_mask,
-                'test_mask': test_mask}
+        if self.label_name is not None:
+            train_mask_name = self.label_name + '_train_mask'
+            val_mask_name = self.label_name + '_val_mask'
+            test_mask_name = self.label_name + '_test_mask'
+        else:
+            train_mask_name = 'train_mask'
+            val_mask_name = 'val_mask'
+            test_mask_name = 'test_mask'
+        return {train_mask_name: train_mask,
+                val_mask_name: val_mask,
+                test_mask_name: test_mask}
+
+class ClassificationProcessor(LabelProcessor):
+    """ Process the label for the classification task.
+    """
+
+    def __call__(self, data):
+        """ Process the label for classification.
+
+        This performs data split on the nodes/edges and generates training/validation/test masks.
+
+        Parameters
+        ----------
+        data : dict of Tensors
+            All data associated with nodes/edges of a node/edge type.
+
+        Returns
+        -------
+        dict of Tensors : it contains the labels as well as training/val/test splits.
+        """
+        assert self.col_name in data, f"The label column {self.col_name} does not exist."
+        label = data[self.col_name]
+        assert np.issubdtype(label.dtype, np.integer) \
+                or np.issubdtype(label.dtype, np.floating), \
```

Comment:
Can it be a float?
There is a conflict between the check and the error message.